WordPress/src/main/java/org/wordpress/android/ui/reader/utils/ReaderUtils.java

```diff
@@ -546,4 +546,12 @@ public static boolean postExists(long blogId, long postId) {
     public static boolean commentExists(long blogId, long postId, long commentId) {
         return ReaderCommentTable.commentExists(blogId, postId, commentId);
     }
+
+    /**
+     * Self-hosted sites have a site id of 0, but we use -1 to indicate a self-hosted site
```

Comment:
This logic seems by trying it out but I'll need to confirm that it applies in all cases.

Reply:
I think this is correct as well, but I agree it'd be good to confirm. :+1:

Reply:
This should be correct since [the `authorBlogId` is actually the `site_ID`](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/models/ReaderPost.java#L279) that exists only for [WPCOM sites](https://developer.wordpress.com/docs/api/1.1/get/read/sites/%24site/posts/%24post_ID/#apidoc-response).

Reply:
I find the logic to be correct as well by trying with the debugger 🤔 

Another option might be to use `post.isWP()` which internally is equal to `!post.isExternal`. I noticed it is used to enable/disable likes & comments post cards, see:
https://github.com/wordpress-mobile/WordPress-Android/blob/0002b4ede131f3bcc8f99596098d3832a81ffa95/WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderPostUiStateBuilder.kt#L439

Although personally I prefer basing the logic on `authorBlogId` because the [issue](https://github.com/wordpress-mobile/WordPress-Android/issues/17826#issue-1557329972) is caused by the value of this field being equal for different sites 🤷 

@antonis wdyt?

Reply:
Good point @ovitrif 👍 
My first though was to use the `isWP` property but I haven't figured out where this value comes from  and it seems inconsistent to me 🤷 
For example for the following sites the `post.isWP` is `false` in my tests:
* https://www.tannr.com
* https://magazine.atavist.com
* https://puny-toucan.jurassic.ninja

and for the https://top-lungfish.jurassic.ninja the value is `true` 😕 
Note that I've created both JN sites at the same time 🤷 
Searching I found this old p5PDj3-Ij-p2 post that describes an issue with the `is_external` value.

Reply:
> For example for the following sites the post.isWP is false in my tests [...]
> Searching I found this old p5PDj3-Ij-p2 post that describes an issue with the is_external value.

Nice find! All more reasons in favor of the current approach 🚀 👍 

I'll resolve this conversation, thank you for the follow-up and the additional info!